scripts/build-transmission.js

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env node
+const os = require('node:os')
+const fs = require('node:fs')
+const path = require('node:path')
+const { spawn } = require('node:child_process')
+
+const env = { ...process.env }
+
+const NPROCESSORS = os.availableParallelism()
+const COMMON_CMAKE_FLAGS = [
+  '-DCMAKE_BUILD_TYPE=Release',
+  '-DENABLE_DAEMON=OFF',
+  '-DENABLE_GTK=OFF',
+  '-DENABLE_QT=OFF',
+  '-DENABLE_UTILS=OFF',
+  '-DENABLE_CLI=OFF',
+  '-DENABLE_TESTS=OFF',
+  '-DENABLE_UTP=ON',
+  '-DENABLE_NLS=OFF',
+  '-DINSTALL_DOC=OFF',
+  '-DINSTALL_LIB=OFF',
+  '-DUSE_SYSTEM_EVENT2=OFF',
+  '-DUSE_SYSTEM_DEFLATE=OFF',
+  '-DUSE_SYSTEM_DHT=OFF',
+  '-DUSE_SYSTEM_MINIUPNPC=OFF',
+  '-DUSE_SYSTEM_NATPMP=OFF',
+  '-DUSE_SYSTEM_UTP=OFF',
+  '-DUSE_SYSTEM_B64=OFF',
+  '-DUSE_SYSTEM_PSL=OFF',
+  '-DWITH_CRYPTO=openssl'
+]
+
+const runCommand = async (command, args = [], options = {}) => {
+  return new Promise((resolve, reject) => {
+    const child = spawn(command, args, { stdio: 'inherit', ...options })
+    child.on('close', resolve)
+    child.on('error', reject)
+  })
+}
```

Comment:
Regarding how it's used in this file, I'd see a cmake function ^^

```suggestion
const cmake = async (args = [], options = {}) => {
  return new Promise((resolve, reject) => {
    const child = spawn("cmake", args, { stdio: 'inherit', ...options })
    child.on('close', resolve)
    child.on('error', reject)
  })
}
```

Reply:
I was thinking this function to be as generic as possible in case we need to run other commands, depending on the os ?
It's true that for now we have only `cmake` commands, but still, wouldn't it be best to let this function generic ? It does not do much, just calling `spawn` wrapped inside a promise.

Reply:
We could also define a intermediate function like so :
```js
const cmake = (args = [], options = {}) => runCommand('cmake',args, options)
```

This way we could keep the generic function and the specific `cmake` function.
Do you think it's better to define a `cmake` function ? I'm rather neutral on this idea, `runCommand` is ok to me, but one might find the `cmake` function more readable ?

Reply:
I'm fine with this approach too ^^